File: packages/burnsystems.flexbg/packages/burnsystems.webserver.umbra/src/BurnSystems.WebServer.Umbra/Umbra.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using BurnSystems.ObjectActivation;
using BurnSystems.Test;
using BurnSystems.WebServer.Dispatcher;
using BurnSystems.WebServer.Responses;

namespace BurnSystems.WebServer.Umbra
{
    public class Umbra
    {
        /// <summary>
        /// Integrates umbra into 
        /// </summary>
        /// <param name="container">Container where umbra is integrated</param>
        public static void Integrate(ActivationContainer container)
        {
            Integrate(container, UmbraConfiguration.Default);
        }

        public static void Integrate(ActivationContainer container, UmbraConfiguration configuration)
        {
            var server = container.Get<Server>();
            Ensure.IsNotNull(server, "No Server in Actioncontainer");

            // AddWebFile(configuration, server, "index.html", "text/html", Resources_Umbra.index);

            // Add core function of umbra
            /*AddWebFile(configuration, server, "js/lib/umbra.js", "text/javascript", Resources_Umbra.js_umbra);
            AddWebFile(configuration, server, "js/lib/umbra.area.js", "text/javascript", Resources_Umbra.js_umbra_area);
            AddWebFile(configuration, server, "js/lib/umbra.dragbar.js", "text/javascript", Resources_Umbra.js_umbra_dragbar);
            AddWebFile(configuration, server, "js/lib/umbra.eventbus.js", "text/javascript", Resources_Umbra.js_umbra_eventbus);
            AddWebFile(configuration, server, "js/lib/umbra.instance.js", "text/javascript", Resources_Umbra.js_umbra_instance);
            AddWebFile(configuration, server, "js/lib/umbra.ribbonbar.js", "text/javascript", Resources_Umbra.js_umbra_ribbonbar);
            AddWebFile(configuration, server, "js/lib/umbra.ribbonbutton.js", "text/javascript", Resources_Umbra.js_umbra_ribbonbutton);
            AddWebFile(configuration, server, "js/lib/umbra.ribbonelement.js", "text/javascript", Resources_Umbra.js_umbra_ribbonelement);
            AddWebFile(configuration, server, "js/lib/umbra.ribbongroup.js", "text/javascript", Resources_Umbra.js_umbra_ribbongroup);
            AddWebFile(configuration, server, "js/lib/umbra.ribbontab.js", "text/javascript", Resources_Umbra.js_umbra_ribbontab);
            AddWebFile(configuration, server, "js/lib/umbra.view.js", "text/javascript", Resources_Umbra.js_umbra_view);
            AddWebFile(configuration, server, "js/lib/umbra.viewpoint.js", "text/javascript", Resources_Umbra.js_umbra_viewpoint);
            AddWebFile(configuration, server, "js/lib/umbra.viewtype.js", "text/javascript", Resources_Umbra.js_umbra_viewtype);
            AddWebFile(configuration, server, "js/lib/umbra.workspace.js", "text/javascript", Resources_Umbra.js_umbra_workspace);*/

            // Add Plugins
            // AddWebFile(configuration, server, "js/lib/umbra.console.js", "text/javascript", Resources_Umbra.js_umbra_console);

            // CSS Style sheet
            // AddWebFile(configuration, server, "css/umbra.css", "text/css", Resources_Umbra.css_umbra);

            // Library stupp
            /*AddWebFile(configuration, server, "js/lib/dateformat.js", "text/javascript", Files.DateFormat);
            AddWebFile(configuration, server, "js/lib/jquery.js", "text/javascript", Files.JQuery);
            AddWebFile(configuration, server, "js/lib/require.js", "text/javascript", Files.Require);*/

            // Initialization
            // AddWebFile(configuration, server, "js/lib/test.js", "text/javascript", Resources_Umbra.js_test);
            // AddWebFile(configuration, server, "js/lib/init.js", "text/javascript", Resources_Umbra.js_init);

            server.Add(
                new RelocationDispatcher(
                    DispatchFilter.ByExactUrl(configuration.WebPath), "index.html"));

            /*server.Add(
                new UmbraDispatcher(
                    DispatchFilter.ByUrl(configuration.WebPath), configuration.WebPath));*/
        }

        private static void AddWebFile(UmbraConfiguration configuration, Server server, string url, string mimeType, string content)
        {
            server.Add(
                new StaticContentResponse(
                    DispatchFilter.ByExactUrl(configuration.WebPath + url), mimeType, content));
        }
    }
}

File: packages/burnsystems.flexbg/packages/burnsystems.webserver.umbra/src/BurnSystems.WebServer.Umbra/htdocs/js/lib/umbra.js
"use strict";

define([
    "umbra.workspace",      // 1
    "umbra.view",           // 2
    "umbra.viewpoint",      // 3
    "umbra.area",           // 4
    "umbra.viewtype",       // 5
    "umbra.ribbonbar",      // 6
    "umbra.ribbonbutton",   // 7
    "umbra.ribbonelement",  // 8
    "umbra.ribbongroup",    // 9
    "umbra.ribbontab",      // 10
    "umbra.eventbus",       // 11
    "umbra.instance",       // 12
    "dateformat"],
    function (
        umbraWorkspaceClass,        // 1
        umbraViewClass,             // 2
        umbraViewPointClass,        // 3
        umbraAreaClass,             // 4
        umbraViewTypeClass,         // 5
        umbraRibbonBarClass,        // 6
        umbraRibbonButtonClass,     // 7
        umbraRibbonElementClass,    // 8
        umbraRibbonGroupClass,      // 9
        umbraRibbonTabClass,        // 10
        umbraEventBusClass,         // 11
        umbraInstance               // 12
        ) {

        ///////////////////////////////////////////
        // Combination of everything
        var result =
            {
                getVersion: function () { return "1.0.0.0"; },

                View: umbraViewClass,
                ViewPoint: umbraViewPointClass,
                WorkSpace: umbraWorkspaceClass,
                RibbonBar: umbraRibbonBarClass,
                RibbonButton: umbraRibbonButtonClass,
                RibbonElement: umbraRibbonElementClass,
                RibbonGroup: umbraRibbonGroupClass,
                RibbonTab: umbraRibbonTabClass,
                Area: umbraAreaClass,
                EventBus: umbraEventBusClass,
                ViewType: umbraViewTypeClass,
                umbra: umbraInstance
            };

        return result;
    });